tools/gloo_wrapper.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+import torch
+import torch.distributed as dist
+
+
+def allgather_sizes(send_data, world_size):
+    """
+    Perform all gather on list lengths, used to compute prefix sums
+    to determine the offsets on each ranks. This is used to allocate
+    global ids for edges/nodes on each ranks.
+    Parameters
+    ----------
+    send_data : Data on which allgather is performed.
+    world_size : No. of processes configured for execution
+    """
+
+    #compute the length of the local data
+    send_length = len(send_data)
+    out_tensor = torch.as_tensor(send_data, dtype=torch.int32)
+    in_tensor = [torch.zeros(send_length, dtype=torch.int32) 
+                    for _ in range(world_size)]
+
+    #all_gather message
+    dist.all_gather(in_tensor, out_tensor)
+
+    #gather sizes in on array to return to the invoking function
+    rank_sizes = [0] * (world_size + 1)
+    count = rank_sizes[0]
+    for i, t in enumerate(in_tensor): 
+        count += t.item()
+        rank_sizes[i+1] = count
+
+    return rank_sizes
+
+
+def alltoall_cpu(rank, world_size, output_tensor_list, input_tensor_list):
```

Comment:
do we want to have `output_tensor_list` as an input argument instead of being return values?